src/v/net/unresolved_address.h

```diff
@@ -34,9 +34,7 @@ class unresolved_address
 
     unresolved_address() = default;
     unresolved_address(
-      ss::sstring host,
-      uint16_t port,
-      inet_family family = ss::net::inet_address::family::INET)
+      ss::sstring host, uint16_t port, inet_family family = std::nullopt)
```

Comment:
Do we ever pass in an explicit `family`?  Maybe we can do away with the argument.

Reply:
Yeah.

* [serialization_rt_test](https://github.com/BenPope/redpanda/blob/88180d9cfc308c87a5055bf14f338bdd25676332/src/v/cluster/tests/serialization_rt_test.cc#L1248-L1249)
* [s3_configuration::make_configuration](https://github.com/BenPope/redpanda/blob/88180d9cfc308c87a5055bf14f338bdd25676332/src/v/cloud_storage_clients/configuration.cc#L95)

It could probably be eliminated, at the cost of making the backports a manul task.

Reply:
It could probably be eliminated, might mean the backport has to be manual and this review more involved.
```diff
diff --git a/src/v/cloud_storage_clients/configuration.cc b/src/v/cloud_storage_clients/configuration.cc
index d5351608a..453b54199 100644
--- a/src/v/cloud_storage_clients/configuration.cc
+++ b/src/v/cloud_storage_clients/configuration.cc
@@ -90,9 +90,7 @@ ss::future<s3_configuration> s3_configuration::make_configuration(
     }
 
     client_cfg.server_addr = net::unresolved_address(
-      client_cfg.uri(),
-      overrides.port ? *overrides.port : default_port,
-      ss::net::inet_address::family::INET);
+      client_cfg.uri(), overrides.port.value_or(default_port));
     client_cfg.disable_metrics = disable_metrics;
     client_cfg.disable_public_metrics = disable_public_metrics;
     client_cfg._probe = ss::make_shared<client_probe>(
@@ -143,9 +141,7 @@ ss::future<abs_configuration> abs_configuration::make_configuration(
     }
 
     client_cfg.server_addr = net::unresolved_address(
-      client_cfg.uri(),
-      overrides.port ? *overrides.port : default_port,
-      ss::net::inet_address::family::INET);
+      client_cfg.uri(), overrides.port.value_or(default_port));
     client_cfg.disable_metrics = disable_metrics;
     client_cfg.disable_public_metrics = disable_public_metrics;
     client_cfg._probe = ss::make_shared<client_probe>(
diff --git a/src/v/cluster/tests/serialization_rt_test.cc b/src/v/cluster/tests/serialization_rt_test.cc
index 304169709..f5e8f5978 100644
--- a/src/v/cluster/tests/serialization_rt_test.cc
+++ b/src/v/cluster/tests/serialization_rt_test.cc
@@ -1243,15 +1243,10 @@ SEASTAR_THREAD_TEST_CASE(serde_reflection_roundtrip) {
         roundtrip_test(data);
     }
     {
-        std::vector<std::optional<ss::net::inet_address::family>> families = {
-          std::nullopt,
-          ss::net::inet_address::family::INET,
-          ss::net::inet_address::family::INET6};
         net::unresolved_address data{
           random_generators::gen_alphanum_string(
             random_generators::get_int(0, 100)),
-          random_generators::get_int<uint16_t>(0, 16000),
-          random_generators::random_choice(families)};
+          random_generators::get_int<uint16_t>(0, 16000)};
         serde_roundtrip_test(data);
         // adl roundtrip doesn't work because family is not serialized
     }
diff --git a/src/v/net/unresolved_address.h b/src/v/net/unresolved_address.h
index 88e51c1ac..993a93b1d 100644
--- a/src/v/net/unresolved_address.h
+++ b/src/v/net/unresolved_address.h
@@ -29,15 +29,14 @@ class unresolved_address
       unresolved_address,
       serde::version<0>,
       serde::compat_version<0>> {
-public:
     using inet_family = std::optional<ss::net::inet_address::family>;
 
+public:
     unresolved_address() = default;
-    unresolved_address(
-      ss::sstring host, uint16_t port, inet_family family = std::nullopt)
+    unresolved_address(ss::sstring host, uint16_t port)
       : _host(std::move(host))
       , _port(port)
-      , _family(family) {}
+      , _family() {}
 
     const ss::sstring& host() const { return _host; }
     uint16_t port() const { return _port; }
```